test/functional/rpc_blockchain.py

```diff
@@ -213,19 +213,6 @@ def check_signalling_deploymentinfo_result(self, gdi_result, height, blockhash,
                 },
                 'active': False
             },
-            'taproot': {
```

Comment:
The other soft forks are marked as burried, rather than completely omitted.

Reply:
Yes, I think this is a reason to remove them as well. Once they are buried, their activation height is a constant. So there is no need to report it dynamically over RPC. Apart from being overkill, I don't see a use case either.

If someone wants to know if the wallet supports $feature, they can use a wallet RPC or the GUI. (This RPC won't help them)
If someone wants to know if the mempool supports $feature, this RPC won't help them either. I think the best they can do is read the release notes to figure out if the mempool supports $feature regardless of blockchain height. Alternatively they can just test if their use case is supported.
If someone read the release notes, but wants to know *over RPC* which version of Bitcoin Core they are running, they can use the appropriate call. 

As mentioned in OP, returning a result for `taproot` here (buried or vb) is neither right or wrong. While the result may reflect the height at which the BIP activated on the network (as mentioned in the BIP as well), it does not reflect the height at which this software enforces the rules. The rules are theoretically enforced since genesis, practically only after witness is allowed in blocks. This is not something that can be explained by returning a JSON.

If you insist on returning a JSON, my preference would be to return `'taproot': True`. Until then, my preference is to remove it, since it is unknown if there is a use case for this (and if there is one) what the use case is.

Reply:
> Yes, I think this is a reason to remove them as well.

I'm not opposed to that, but the inconsistency doesn't help review. Why not set it to `burried` now and make another PR to drop all buried soft forks from the RPC, using the above reasoning. Such a followup would be an RPC-only change.

Reply:
Ok, my suggestion to remove the others as well was a bit over eager. They are still used and the activation height actually affects code paths. In fact the tests use `-testactivationheight` to mock the activation height. Though, this isn't true for `taproot`.

Reply:
Also, I already tried to bury taproot in #23505. Though there wasn't much/enough appetite to get it in before the 23.0 feature freeze. As burying is a breaking RPC change and we already had one (getdeploymentinfo) in 23.0, it could have been bundled with that. However, now that 23.0 is out, it would be a breaking change in 24.0 to bury it. Then, if it is decided to remove, it will be another breaking change.

Reply:
Why is it a breaking change? A correct parser should look for `taproot` and then check the `type` field. It should be able to handle `buried`. If an implementation can't handle that, it might also break when `taproot` is missing entirely.